### PR TITLE
remove blockexit dependency on global.errors

### PR DESCRIPTION
**Pull request by WalterBright**
Also start modernizing the `BE` enum.

### CODE REVIEW

File: src/ddmd/blockexit.d [hunk 1]
```diff
@@ -36,34 +36,38 @@ import ddmd.visitor;
...
-    BEerrthrow = 0x80,
-    BEany      = (BEfallthru | BEthrow | BEreturn | BEgoto | BEhalt),
+    none      = 0,
+    fallthru  = 1,
+    _throw    = 2,
+    _return   = 4,
+    _goto     = 8,
+    halt      = 0x10,
+    _break    = 0x20,
+    _continue = 0x40,
```
Comment (mathias-lang-sociomantic) [inline src/ddmd/blockexit.d]: https://dlang.org/dstyle.html#naming_keywords (underscore should be appended, not prepended)

File: src/ddmd/blockexit.d [hunk 2]
```diff
@@ -36,34 +36,38 @@ import ddmd.visitor;
...
+alias BEnone     = BE.none;
+alias BEfallthru = BE.fallthru;
+alias BEthrow    = BE._throw;
+alias BEreturn   = BE._return;
+alias BEgoto     = BE._goto;
+alias BEhalt     = BE.halt;
+alias BEbreak    = BE._break;
+alias BEcontinue = BE._continue;
+alias BEerrthrow = BE.errthrow;
+alias BEany      = BE.any;
```
Comment (mathias-lang-sociomantic) [inline src/ddmd/blockexit.d]: Is there any plan to get rid of those eventually ?

Comment (WalterBright) [reply src/ddmd/blockexit.d]: yes

File: src/ddmd/blockexit.d [hunk 3]
```diff
@@ -36,34 +36,38 @@ import ddmd.visitor;
...
- * Only valid after semantic analysis
+ * Determine mask of ways that a statement can exit.
+ *
+ * Only valid after semantic analysis.
  * Params:
+ *   s = statement to check for block exit status
+ *   func = function that statement s is in
  *   mustNotThrow = generate an error if it throws
  * Returns:
  *   BExxxx
```
Comment (mathias-lang-sociomantic) [inline src/ddmd/blockexit.d]: Maybe that part of the doc can be updated as well since we have a proper enum now

File: src/ddmd/blockexit.d [hunk 4]
```diff
@@ -94,7 +98,7 @@ int blockExit(Statement s, FuncDeclaration func, bool mustNotThrow)
 
         override void visit(ErrorStatement s)
         {
-            result = BEany;
+            result = BE.none;
```
Comment (mathias-lang-sociomantic) [inline src/ddmd/blockexit.d]: That doesn't look like a refactor...

Comment (WalterBright) [reply src/ddmd/blockexit.d]: This is completely trivial. My complaint with mixing refactoring with semantic changes is when there'd be a hundred lines of code moved about that completely obfuscated what the actual fix was. That is not the case here.

File: src/ddmd/blockexit.d [hunk 5]
```diff
@@ -36,34 +36,38 @@ import ddmd.visitor;
...
+    none      = 0,
+    fallthru  = 1,
+    throw_    = 2,
+    return_   = 4,
+    goto_     = 8,
+    halt      = 0x10,
+    break_    = 0x20,
+    continue_ = 0x40,
+    errthrow  = 0x80,
+    any       = (fallthru | throw_ | return_ | goto_ | halt),
```
Comment (andralex) [inline src/ddmd/blockexit.d]: so nice

File: src/ddmd/blockexit.d [hunk 6]
```diff
@@ -36,34 +36,38 @@ import ddmd.visitor;
...
+alias BEnone     = BE.none;
+alias BEfallthru = BE.fallthru;
+alias BEthrow    = BE.throw_;
+alias BEreturn   = BE.return_;
+alias BEgoto     = BE.goto_;
+alias BEhalt     = BE.halt;
+alias BEbreak    = BE.break_;
+alias BEcontinue = BE.continue_;
+alias BEerrthrow = BE.errthrow;
+alias BEany      = BE.any;
```
Comment (andralex) [inline src/ddmd/blockexit.d]: meh vertical alignment sux

Comment (andralex) [reply src/ddmd/blockexit.d]: I assume these are for backward compat and will be phased out?

**Review by andralex — APPROVED** [review]

File: src/ddmd/blockexit.d (hunk 6 above)
Comment (WalterBright) [reply src/ddmd/blockexit.d]: yes

**Pull request merged**